### PR TITLE
Added compose prod and dockerfile prod

### PR DESCRIPTION
**Pull request by faisalisafk**
Fixes #

- [ ] Does not depend on other PRs
- [x] Depends on other PR being merged first: #283

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [ ] Improvement
- [x] New feature
- [ ] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?
Added Docker file and docker compose file to create volume and implement multi stage build
## What was done
- Added Dockerfile and docker-compose.yml to create container with a volume binding of todo app
- Added Dockerfile.prod and docker-compose.prod.yml to create multi stage container of build file and nginx
## How to test

- [ ] Needs regression testing (add QA needed label)

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

## Reminder list

- Determine whether REST API Documentation should be updated
- Make an assessment on whether unit tests should be updated and/or added
- Ensure follow up tasks are linked to this PR

QA - if regression testing was requested:

- [ ] QA has verified

### CODE REVIEW

File: faisal/todo/Dockerfile [hunk 1]
```diff
@@ -2,11 +2,14 @@
 FROM node:alpine
 #makes a new working directory in the linux distribution
 WORKDIR /faisal-code
+#Copying package.json and lock.json required to run the project dependency
+COPY package.json .
+COPY package-lock.json .
```
Comment (musfique-m) [inline faisal/todo/Dockerfile:7]: Please use wildcard here to copy both files in one go. This way we can reduce the number of layers.

File: faisal/todo/Dockerfile.prod [hunk 2]
```diff
@@ -0,0 +1,12 @@
+FROM node:alpine as build
+WORKDIR /faisal-code
+COPY package.json .
+COPY package-lock.json .
```
Comment (musfique-m) [inline faisal/todo/Dockerfile.prod:4]: Please also use wildcard here.

File: faisal/docker-compose.prod.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+version: "3.9" 
+services:
+    react-nginx-app:
+      stdin_open: true
```
Comment (musfique-m) [inline faisal/docker-compose.prod.yml:4]: Why are we running our production container in interactive mode?

**Review by musfique-m — CHANGES_REQUESTED** [review]

File: faisal/todo/Dockerfile (hunk 1 above)
Comment (faisalisafk) [reply faisal/todo/Dockerfile]: Added wildcards in [c1482ff](https://github.com/CraftsmenLtd/coe-batch-q322/pull/295/commits/c1482ff80747ef39ca4e5c28aace8964c5bb128b)

File: faisal/todo/Dockerfile.prod (hunk 2 above)
Comment (faisalisafk) [reply faisal/todo/Dockerfile.prod]: Added wildcards in [c1482ff](https://github.com/CraftsmenLtd/coe-batch-q322/pull/295/commits/c1482ff80747ef39ca4e5c28aace8964c5bb128b)

File: faisal/docker-compose.prod.yml (hunk 3 above)
Comment (faisalisafk) [reply faisal/docker-compose.prod.yml]: I have removed interactive mode in [c1482ff](https://github.com/CraftsmenLtd/coe-batch-q322/pull/295/commits/c1482ff80747ef39ca4e5c28aace8964c5bb128b) .

**Review by musfique-m — APPROVED** [review]
Looks good.

**Pull request merged**